src/target/riscv/riscv-013.c

```diff
@@ -498,6 +498,23 @@ static void increase_dmi_busy_delay(struct target *target)
 	dtmcontrol_scan(target, DTM_DTMCS_DMIRESET, NULL /* discard result */);
 }
 
+static void decrease_reset_delays_wait(struct target *target, size_t finished_scans)
+{
+	RISCV_INFO(r);
+	if (r->reset_delays_wait < 0) {
+		assert(r->reset_delays_wait == -1);
+		return;
+	}
+	if ((size_t)r->reset_delays_wait >= finished_scans) {
+		r->reset_delays_wait -= finished_scans;
+		return;
+	}
+	r->reset_delays_wait = -1;
+	LOG_TARGET_DEBUG(target, "reset_delays_wait done");
```

Comment:
If you'd like, the log message can be made more clear:

```suggestion
	LOG_TARGET_DEBUG(target, "resetting learned delays (reset_delays_wait counter expired)");
```

Reply:
Addressed.